toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,328 @@
+	// bootif=01-<MAC Address>, where each byte value of the MAC address is separated
+	// by dashes instead of colons. Therefore, we're reading from the 4th spot of the
+	// string to obtain the MAC address and then replace the dashes with colons
+	deviceAddr = strings.ReplaceAll(bootifValue[macAddressStartIndex:len(bootifValue)], "-", ":")
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device, err = findBootIfValue()
+		if err != nil {
+			logger.Log.Errorf("Failed to read bootif value from /proc/cmdline")
+			return
+		}
+	}
+
+	for _, iface := range ifaces {
+		ifaceHardwareAddr := strings.TrimSpace(iface.HardwareAddr.String())
+		if networkData.Device == iface.Name || strings.EqualFold(ifaceHardwareAddr, networkData.Device) {
+			deviceName = iface.Name
+			return
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	matchSection := fmt.Sprintf("[Match]\nName=%s\n", deviceName)
+	err = file.Append(matchSection, fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	var networkSection strings.Builder
+
+	// Write network section tag
+	networkSection.WriteString("[Network]\n")
+
+	// Currently only supports static IP setting
+	// Write IP and netmask
+	if networkData.NetMask != "" && networkData.Ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		networkMask, _ := stringMask.Size()
+		networkSection.WriteString(fmt.Sprintf("Address=%s/%d\n", networkData.Ip, networkMask))
+	}
+
+	// Write Gateway
+	if networkData.GateWay != "" {
+		networkSection.WriteString(fmt.Sprintf("Gateway=%s\n", networkData.GateWay))
+	}
+
+	// Write Nameserver
+	for _, nameserver := range networkData.NameServers {
+		networkSection.WriteString(fmt.Sprintf("DNS=%s\n", nameserver))
+	}
+
+	err = file.Append(networkSection.String(), fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Network] section: %s", err)
+	}
+
+	return
+}
+
+func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Network, deviceName string) (err error) {
+	const (
+		networkFileDir = "/etc/systemd/network"
+		filePrefix     = "10"
+	)
+
+	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/systemd/network/")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/systemd/network/: no such path or directory")
+		return
+	}
+
+	logger.Log.Debugf("Start creating network file")
+
+	networkFilePath := fmt.Sprintf("%s/%s-%s-%s.network", networkFileDir, filePrefix, networkData.BootProto, deviceName)
+	exists, err := file.PathExists(networkFilePath)
```

Comment:
I could be wrong here it seems the value of "err" from lines 267/270 are overwritten here before being checked. Should we check for the error before overwritting it here?

Reply:
I don't think we're overwriting here since at this point right before we're assigning values to err, err should be nil. Back in line 267 and 270, err is set and then the function immediately returns.